prusti-viper/src/encoder/mir/specifications/interface.rs

```diff
@@ -63,26 +65,29 @@ impl<'tcx> SpecQuery<'tcx> {
                     call_substs: new_substs,
                 })
             }
-            PureOrTrustedCheck(_, _) => PureOrTrustedCheck(new_def_id, new_substs),
+            GetProcKind(_, _) => GetProcKind(new_def_id, new_substs),
             FetchSpan(_) => FetchSpan(new_def_id),
         }
     }
 }
 
 pub(crate) trait SpecificationsInterface<'tcx> {
+    // TODO abstract-predicates: Maybe this should be deleted (and ProcedureSpecificationKind::is_pure)
```

Comment:
I agree, it might be confusing that predicates also return `true` here. How many usages are there?

Reply:
I found 8 in total.
We maybe could just rename the method to `must_be_encoded_as_pure` or something similar, afaik this is how we interact with this method.